src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -862,24 +860,6 @@ private void EmitMemberGet(MemberInfo member, Type objectType)
             }
         }
 
-        private static bool TryGetRawConstantValue(FieldInfo fi, out object value)
-        {
-            // TODO: It looks like GetRawConstantValue is not available at the moment, use it when it comes back.
-            //value = fi.GetRawConstantValue();
-            //return true;
-
-            try
-            {
-                value = fi.GetValue(obj: null);
-                return true;
-            }
-            catch
-            {
-                value = null;
```

Comment:
I think that's in the same category as a stream that returns true for `CanRead` and throws NotSupported on `Read()`; if you're returning true for `IsLiteral` and throw on `GetRawConstantValue` you deserve to lose and the sooner things start throwing when they use your code the sooner you can fix it.